sdk/identity/azure-identity/src/main/java/com/azure/identity/SharedTokenCacheCredential.java

```diff
@@ -41,31 +43,44 @@
      * @param clientId the client ID of the application
      * @param identityClientOptions the options for configuring the identity client
      */
-    SharedTokenCacheCredential(String username, String clientId, IdentityClientOptions identityClientOptions) {
+    SharedTokenCacheCredential(String username, String clientId, String tenantId,
```

Comment:
I don't see `tenantId` used to filter accounts, does it magically happen elsewhere?

Reply:
Correct. This tenant ID only appears on the authority URL. The tenant ID info in the homeAccountId field is always going to be the home tenant ID for the account. You can get an access token for this tenant even if the refresh token is acquired for a different tenant (home tenant) - this is not a bug. The bug that we reported to MSAL is that you can get an access token for this tenant *even if* you don't have access to this tenant.

Since the tenant ID in the homeAccountId in the token cache is always going to be the home tenant, there's no filtering to do there.